frontend/task/RecorderControlsTask.tsx

```diff
@@ -0,0 +1,167 @@
+import React from "react";
+import {Button, ButtonGroup, Icon, Slider} from "@blueprintjs/core";
+import {formatTime} from "../common/utils";
+import {ActionTypes as RecorderActionTypes} from "../recorder/actionTypes";
+import {ActionTypes as PlayerActionTypes} from "../player/actionTypes";
+import {connect} from "react-redux";
+import {AppStore} from "../store";
+import {getPlayerState} from "../player/selectors";
+import {getRecorderState} from "../recorder/selectors";
+import {RecorderStatus} from "../recorder/store";
+import {MemoryUsage} from "../recorder/MemoryUsage";
+
+interface RecorderControlsTaskStateToProps {
+    getMessage: Function,
+    canRecord: boolean,
+    canPlay: boolean,
+    canPause: boolean,
+    canStop: boolean,
+    canStep: boolean,
+    isPlayback: boolean,
+    playPause: 'play' | 'pause',
+    position: number,
+    duration: number,
+    recorderStatus: RecorderStatus
+}
+
+function mapStateToProps (state: AppStore): RecorderControlsTaskStateToProps {
+    const getMessage = state.getMessage;
+    const recorder = getRecorderState(state);
+    const recorderStatus = recorder.status;
+    const isPlayback = recorderStatus === RecorderStatus.Paused;
+    let canRecord, canPlay, canPause, canStop, canStep, position, duration, playPause;
+
+    if (isPlayback) {
+        const player = getPlayerState(state);
+        const isReady = player.isReady;
+        const isPlaying = player.isPlaying;
+        canPlay = canStop = canRecord = canStep = isReady && !isPlaying;
+        canPause = isReady && isPlaying;
+        playPause = isPlaying ? 'pause' : 'play';
+        position = player.audioTime;
+        duration = player.duration;
+    } else {
+        canRecord = /ready|paused/.test(recorderStatus);
```

Comment:
When we first open the recorder and we didn't give authorization to record the microphone yet, the record button is disabled. That could be confusing for first-time users.

Reply:
I think an error should occur as soon as they open the recorder panel if the preparation of the recording doesn't work. I've added this.